pkg/distributor/push.go

```diff
@@ -277,6 +288,44 @@ func isRemoteWrite2(r *http.Request) (bool, error) {
 	return false, nil
 }
 
+// Consts from https://github.com/prometheus/prometheus/blob/main/storage/remote/stats.go
+const (
+	rw20WrittenSamplesHeader    = "X-Prometheus-Remote-Write-Samples-Written"
+	rw20WrittenHistogramsHeader = "X-Prometheus-Remote-Write-Histograms-Written"
+	rw20WrittenExemplarsHeader  = "X-Prometheus-Remote-Write-Exemplars-Written"
+)
+
+func contextWithWriteResponseStats(ctx context.Context) context.Context {
+	return context.WithValue(ctx, pushResponseStatsContextKey, &promRemote.WriteResponseStats{})
+}
+
+func addWriteResponseStats(ctx context.Context, w http.ResponseWriter) error {
+	rsValue := ctx.Value(pushResponseStatsContextKey)
+	if rsValue == nil {
+		return errors.New("error write response stats not found in context")
+	}
+	prs, ok := rsValue.(*promRemote.WriteResponseStats)
+	if !ok {
+		return errors.New("error write response stats not of type *promRemote.WriteResponseStats")
```

Comment:
Don't you just mean "write response stats"?
```suggestion
		return errors.New("write response stats not of type *promRemote.WriteResponseStats")
```

Reply:
i replaced "error" with "remote"